src/Security/LogOutAuthenticationHandler.php

```diff
@@ -26,6 +27,11 @@ public function logIn(Member $member, $persistent = false, HTTPRequest $request
 
     public function logOut(HTTPRequest $request = null)
     {
+        if ($request == null && !Controller::has_curr()) {
+            throw new InvalidArgumentException("No HTTPRequest is provided and there is no controller either.");
+        }
+        $request = Controller::curr()->request;
```

Comment:
Same feedback as with the other instance of this logic.

```suggestion
        // Fall back to retrieving request from current Controller if available
        if ($request === null) {
            if (!Controller::has_curr()) {
                throw new InvalidArgumentException("Authentication with SessionManager enabled requires an active HTTPRequest.");
            }
            
            $request = Controller::curr()->getRequest();
        }
```